grails-app/services/com/unifina/service/EthereumIntegrationKeyService.groovy

```diff
@@ -109,15 +112,20 @@ class EthereumIntegrationKeyService {
 		IntegrationKey.findAllByServiceAndUser(IntegrationKey.Service.ETHEREUM, user)
 	}
 
-	SecUser getOrCreateFromEthereumAddress(String address){
+	SecUser getOrCreateFromEthereumAddress(String address) {
 		IntegrationKey key = IntegrationKey.findByIdInServiceAndService(address, IntegrationKey.Service.ETHEREUM_ID)
-		if(key==null){
-			SecUser user = new SecUser(
-				username: address,
-				password: generatePassword(16),
-				name: address,
-				timezone: "UTC"
-			).save(failOnError: true, flush: true, validate: false)
+		if (key == null) {
+			Calendar now = Calendar.getInstance()
+			TimeZone timeZone = now.getTimeZone()
+			SecUser user = userService.createUser([
+				username       : address,
+				password       : RandomStringUtils.random(32),
+				name           : address,
+				timezone       : timeZone.getDisplayName(),
```

Comment:
This will just return server timezone. We can't know the user timezone: the previous version where you gave UTC here was ok. We will soon get rid of server-side timestamp rendering anyway (#511), so no big deal!